Runtime/Client/EndPointClass.cs

```diff
@@ -23,5 +24,25 @@ public EndPointClass(string endPoint, LootLockerHTTPMethod httpMethod, LootLocke
             this.httpMethod = httpMethod;
             this.callerRole = callerRole;
         }
+
+        public string WithPathParameter(object arg0)
+        {
+            return string.Format(endPoint, WebUtility.UrlEncode(arg0.ToString()));
+        }
+
+        public string WithPathParameters(object arg0, object arg1)
+        {
+            return string.Format(endPoint, WebUtility.UrlEncode(arg0.ToString()), WebUtility.UrlEncode(arg1.ToString()));
+        }
+
+        public string WithPathParameters(object arg0, object arg1, object arg2)
+        {
+            return string.Format(endPoint, WebUtility.UrlEncode(arg0.ToString()), WebUtility.UrlEncode(arg1.ToString()), WebUtility.UrlEncode(arg2.ToString()));
+        }
+
+        public string WithPathParameters(object arg0, object arg1, object arg2, object arg3)
+        {
+            return string.Format(endPoint, WebUtility.UrlEncode(arg0.ToString()), WebUtility.UrlEncode(arg1.ToString()), WebUtility.UrlEncode(arg2.ToString()), WebUtility.UrlEncode(arg3.ToString()));
+        }
```

Comment:
This could be one function and take an array or list instead.
Slightly more cumbersome in the functions, but makes it scale better.
I'm fine with it like this, just wanted to mention it in case you didn't think about it.

Reply:
I did. I just mimicked what string.Format did and they had this structure up to a point to make the call sites more readable.